monai/apps/auto3dseg/algorithms/templates/dints/scripts/validate.py

```diff
@@ -0,0 +1,236 @@
+import logging
+import os
+import sys
+from typing import Optional, Sequence, Union
+
+import numpy as np
+import torch
+import yaml
+
+import monai
+from monai import transforms
+from monai.bundle import ConfigParser
+from monai.bundle.scripts import _pop_args, _update_args
+from monai.data import ThreadDataLoader, decollate_batch
+from monai.inferers import sliding_window_inference
+from monai.metrics import compute_meandice
+
+
+def run(config_file: Optional[Union[str, Sequence[str]]] = None, **override):
+    logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+
+    _args = _update_args(config_file=config_file, **override)
+    config_file_ = _pop_args(_args, "config_file")[0]
+
+    parser = ConfigParser()
+    parser.read_config(config_file_)
+    parser.update(pairs=_args)
+
+    data_file_base_dir = parser.get_parsed_content("data_file_base_dir")
+    data_list_file_path = parser.get_parsed_content("data_list_file_path")
+    fold = parser.get_parsed_content("fold")
+    num_sw_batch_size = parser.get_parsed_content("num_sw_batch_size")
+    output_classes = parser.get_parsed_content("output_classes")
+    overlap_ratio = parser.get_parsed_content("overlap_ratio")
+    patch_size_valid = parser.get_parsed_content("patch_size_valid")
+    softmax = parser.get_parsed_content("softmax")
+
+    ckpt_name = parser.get_parsed_content("validate")["ckpt_name"]
+    output_path = parser.get_parsed_content("validate")["ouptut_path"]
+    save_mask = parser.get_parsed_content("validate")["save_mask"]
+
+    if not os.path.exists(output_path):
+        os.makedirs(output_path, exist_ok=True)
+
+    infer_transforms = parser.get_parsed_content("transforms_infer")
+    validate_transforms = transforms.Compose(
+        [
+            infer_transforms,
+            transforms.LoadImaged(keys="label"),
+            transforms.EnsureChannelFirstd(keys="label"),
+            transforms.EnsureTyped(keys="label"),
+        ]
+    )
+
+    with open(data_list_file_path) as f:
+        json_data = json.load(f)
+
+    list_valid = []
+    for item in json_data["training"]:
+        if item["fold"] == fold:
+            item.pop("fold", None)
+            list_valid.append(item)
+
+    files = []
+    for _i in range(len(list_valid)):
+        str_img = os.path.join(data_file_base_dir, list_valid[_i]["image"])
+        str_seg = os.path.join(data_file_base_dir, list_valid[_i]["label"])
+
+        if (not os.path.exists(str_img)) or (not os.path.exists(str_seg)):
+            continue
+
+        files.append({"image": str_img, "label": str_seg})
+
+    val_files = files
+
+    val_ds = monai.data.Dataset(data=val_files, transform=validate_transforms)
+    val_loader = ThreadDataLoader(val_ds, num_workers=2, batch_size=1, shuffle=False)
+
+    device = torch.device("cuda:0")
+    torch.cuda.set_device(device)
+
+    model = parser.get_parsed_content("network")
+    model = model.to(device)
+    model = torch.nn.SyncBatchNorm.convert_sync_batchnorm(model)
```

Comment:
Same as my comment in infer.py, SyncBatchNorm caused some issue when I trained with one GPU